test/envtest/admission_webhook_envtest_test.go

```diff
@@ -656,14 +656,26 @@ func TestAdmissionWebhook_KongClusterPlugins(t *testing.T) {
 			errorContains: "Change on secret will generate invalid configuration for KongClusterPlugin",
 		},
 	}
+
+	const (
+		waitTime = 30 * time.Second
+		tickTime = 100 * time.Millisecond
+	)
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			require.NoError(t, ctrlClient.Create(ctx, tc.secretBefore))
 			t.Cleanup(func() {
 				require.NoError(t, ctrlClient.Delete(ctx, tc.secretBefore))
 			})
 
-			require.NoError(t, ctrlClientGlobal.Create(ctx, tc.kongClusterPlugin))
+			// NOTE: We create the KongClusterPlugin with 'eventually' to avoid
+			// flaky errors like:
+			// admission webhook "kongclusterplugins.validation.ingress-controller.konghq.com" denied
+			// the request: could not parse plugin configuration: Secret "cluster-conf-secret-valid-patch" not found
+			require.EventuallyWithT(t, func(t *assert.CollectT) {
+				require.NoError(t, ctrlClientGlobal.Create(ctx, tc.kongClusterPlugin))
```

Comment:
```suggestion
				assert.NoError(t, ctrlClientGlobal.Create(ctx, tc.kongClusterPlugin))
```
Otherwise, you'll get a panic in the case of a failure of the assertion instead of graceful retry with an error collected